cbqueryx/queryrespreader.go

```diff
@@ -203,7 +203,43 @@ func (r *queryRespReader) parseError(errJson *queryErrorJson) *ServerError {
 	}
 }
 
-func createResourceError(msg string, cause error) ResourceError {
+func createErrorWithResourceDetail(msg string, cause error) ErrorWithResourceDetail {
+	err := ErrorWithResourceDetail{
+		Cause: cause,
+	}
+
+	if err.Cause == ErrScopeNotFound || err.Cause == ErrCollectionNotFound {
+		parseResourceNotFoundMsg(&err, msg)
+	}
+
+	if err.Cause == ErrAuthenticationFailure {
+		parseAuthFailureMsg(&err, msg)
+	}
+
+	if errors.Is(err.Cause, ErrServerInvalidArg) {
```

Comment:
Why do we use a mix of == and .Is?

Reply:
Because with the InvalidArg error the cause is `ErrInternalServerError` which in turn is caused by the `InvalidArgErr` so we need to use Is here to unwrap down to the InvalidArgErr, whereas the other cases don't have this error nesting

Reply:
I've changed them all to errors.Is for consistency and robustness to future wrapping